src/nvim/drawscreen.c

```diff
@@ -1394,8 +1395,11 @@ static void win_update(win_T *wp, DecorProviders *providers)
   if (type >= UPD_NOT_VALID) {
     // TODO(bfredl): should only be implied for CLEAR, not NOT_VALID!
```

Comment:
FWIW I quickly tested and if bfredl ever gets around to this TODO it seems the below change will need it's own `UPD_NOT_VALID` clause to keep tests passing.